netbox/vpn/filtersets.py

```diff
@@ -0,0 +1,240 @@
+import django_filters
+from django.db.models import Q
+from django.utils.translation import gettext as _
+
+from dcim.models import Interface
+from ipam.models import IPAddress
+from netbox.filtersets import NetBoxModelFilterSet
+from tenancy.filtersets import TenancyFilterSet
+from utilities.filters import ContentTypeFilter, MultiValueCharFilter, MultiValueNumberFilter
+from virtualization.models import VMInterface
+from .choices import *
+from .models import *
+
+__all__ = (
+    'IKEPolicyFilterSet',
+    'IKEProposalFilterSet',
+    'IPSecPolicyFilterSet',
+    'IPSecProfileFilterSet',
+    'IPSecProposalFilterSet',
+    'TunnelFilterSet',
+    'TunnelTerminationFilterSet',
+)
+
+
+class TunnelFilterSet(NetBoxModelFilterSet, TenancyFilterSet):
+    status = django_filters.MultipleChoiceFilter(
+        choices=TunnelStatusChoices
+    )
+    encapsulation = django_filters.MultipleChoiceFilter(
+        choices=TunnelEncapsulationChoices
+    )
+    ipsec_profile_id = django_filters.ModelMultipleChoiceFilter(
+        queryset=IPSecProfile.objects.all(),
+        label=_('IPSec profile (ID)'),
+    )
+    ipsec_profile = django_filters.ModelMultipleChoiceFilter(
+        field_name='ipsec_profile__name',
+        queryset=IPSecProfile.objects.all(),
+        to_field_name='name',
+        label=_('IPSec profile (name)'),
+    )
+
+    class Meta:
+        model = Tunnel
+        fields = ['id', 'name', 'tunnel_id']
+
+    def search(self, queryset, name, value):
```

Comment:
Would it be useful to add a has_terminations like on cables?

Reply:
I don't think we have a `has_terminations` filter anywhere. We have an `unterminated` filter on CableFilterSet, but that's more for checking for missing terminations, which doesn't really apply to tunnels IMO.